cli/tools/doc.rs

```diff
@@ -54,51 +60,33 @@ pub async fn print_docs(
           },
         )
         .await;
-      let doc_parser = doc::DocParser::new(
-        graph,
-        doc_flags.private,
-        analyzer.as_capturing_parser(),
-      );
+      let doc_parser =
+        doc::DocParser::new(&graph, doc_flags.private, capturing_parser)?;
       doc_parser.parse_module(&source_file_specifier)?.definitions
     }
     DocSourceFileFlag::Path(source_file) => {
-      let file_fetcher = factory.file_fetcher()?;
       let module_graph_builder = factory.module_graph_builder().await?;
       let maybe_lockfile = factory.maybe_lockfile();
-      let parsed_source_cache = factory.parsed_source_cache();
 
       let module_specifier =
         resolve_url_or_path(&source_file, cli_options.initial_cwd())?;
 
-      // If the root module has external types, the module graph won't redirect it,
-      // so instead create a dummy file which exports everything from the actual file being documented.
-      let root_specifier =
-        resolve_path("./$deno$doc.ts", cli_options.initial_cwd()).unwrap();
-      let root = File {
-        maybe_types: None,
-        media_type: MediaType::TypeScript,
-        source: format!("export * from \"{module_specifier}\";").into(),
-        specifier: root_specifier.clone(),
-        maybe_headers: None,
-      };
-
-      // Save our fake file into file fetcher cache.
-      file_fetcher.insert_cached(root);
```

Comment:
Sweet that this funky bit is gone!

Reply:
Yeah, deno_doc got smarter here: https://github.com/denoland/deno_doc/blob/eaed9ca50825c2f36ca0b3a5c26d1059af72decd/src/parser.rs#L241-L259

Reply:
That actually probably fixes some bugs too.